assets/js/hook.js

```diff
@@ -0,0 +1,270 @@
+////////////////////////////////////////////////////////////////////////////////////////
+// Hook
+////////////////////////////////////////////////////////////////////////////////////////
+class Hook {
+    constructor(id, node, caption = '', type = 'default', io = 'out',
+                links_options = {}, expects = null, provides = null) {
```

Comment:
Could add a plain text description of what is a Hook, conceptually.

Reply:
I suggest:
> Hooks represent the inputs and the outputs of a node. To define a connection between two nodes, one must specify its source (ie. one output port of a node) and its target (ie. one input port of a node). Hook _io_ can be _i_ if the port  in an _entry/input to the node_ or _o_ if the port is an _output/exit from the node_.